node/node.go

```diff
@@ -410,6 +410,10 @@ func (node *AlgorandFullNode) Stop() {
 	defer func() {
 		node.mu.Unlock()
 		node.waitMonitoringRoutines()
+		// we want to shut down the compactCert last, since the oldKeyDeletionThread might depend on it when making the
+		// call to LatestSigsFromThisNode.
+		node.compactCert.Shutdown()
+		node.compactCert = nil
```

Comment:
waitMonitoringRoutines makes sure that the oldKeyDeletionThread has terminated.
Why setting node.compactCert is necessary? Or this is just for extra layer of protection?

Reply:
Setting the `node.compactCert` to nil was to ensure that no-one would attempt to re-use this.